api/src/devcomp/domain/usecases/terminate-passage.js

```diff
@@ -1,13 +1,20 @@
 import { NotFoundError } from '../../../shared/domain/errors.js';
 import { PassageDoesNotExistError, PassageTerminatedError } from '../errors.js';
+import { PassageTerminatedEvent } from '../models/passage-events/passage-events.js';
 
-async function terminatePassage({ passageId, passageRepository }) {
+async function terminatePassage({ passageId, passageRepository, passageEventRepository }) {
   const passage = await _getPassage({ passageId, passageRepository });
   if (passage.terminatedAt) {
     throw new PassageTerminatedError();
   }
   passage.terminate();
-  return passageRepository.update({ passage });
+  const terminatedPassage = await passageRepository.update({ passage });
+  const event = new PassageTerminatedEvent({
+    passageId: terminatedPassage.id,
+    occurredAt: terminatedPassage.terminatedAt,
```

Comment:
Hmmmm en première itération je serai plutôt pour initier le `occuredAt` avant la moindre action du usecase, c'est un peu technique dans le usecase, mais ça devrait produire un résultat plus proche de ce qu'on aura dans le futur (`occuredAt` < `createdAt`). Ou bien on a un intérêt à mettre la même valeur pour reconnaitre plus facilement ces premiers événements ?

Reply:
@er-lim propose de récupérer l'`occuredAt` de manière générique côté router pour set cet `occurredAt`🙂